backends/edk2-compat/edk2-svc-write.c

```diff
@@ -31,109 +29,119 @@ static int parseArgs(int argc, char *argv[], struct Arguments *args);
 int performWriteCommand(int argc, char* argv[])
 {
 	int rc;
+	char *programName = NULL;
 	struct Arguments args = {	
 		.helpFlag = 0, .inpValid = 0, 
 		.pathToSecVars = NULL, .inFile = NULL, .varName = NULL
 	};
+	// go back two arguments to get program name and subcommand
+	argc += 2;
+	argv -= 2;
+	// we could iterate to just the subcommand but then the usage would look a bit strange
+	// argp will only allow one string in argv as the program name, we want it to be 'secvarctl write' 
+	programName = malloc(strlen(argv[0]) + strlen(argv[1]) + strlen(" ") + 1);
+	if (!programName) { 
+		prlog(PR_ERR, "ERROR: failed to allocate memory\n");
+		return ALLOC_FAIL;
+	}
+	programName = strcpy(programName, argv[0]);
+	programName = strcat(programName, " ");
+	programName = strcat(programName, argv[1]);
+	argc--;
+	argv++;
+	argv[0] = programName;
+
+	struct argp_option options[] = 
+	{
+		{"verbose", 'v', 0, 0, "print more verbose process information"},
+		{"force", 'f', 0, 0, "force update, skips validation of file"},
+		{"path_to_vars", 'p', "PATH" ,0, "looks for .../<var>/update file in PATH, default is " SECVARPATH},
```

Comment:
Personal nit: underscores are a bit weird in argument long names, consider replacing with hyphens, or rewording to not need them? e.g. `--path-to-vars` or `--varpath`

Reply:
Good call. fixing this....